### PR TITLE
[svg][iOS][Android] Upgrade `react-native-svg@12.1.1 ➡️ 12.3.0`

### PR DESCRIPTION
**Pull request by bbarthec**
# Why

Resolves [ENG-2603](https://linear.app/expo/issue/ENG-2603/react-native-screens-380-3100)

[Full CHANGELOG `v12.1.1...v12.3.0`](https://github.com/react-native-svg/react-native-svg/compare/v12.1.1...v12.3.0)

# How

1. `et uvm --module "react-native-svg" --commit "v12.3.0" --target "expo-go"`
2. `yarn`

# Test Plan

Tested `SVG` screen:
- on Android and iOS using `bare-expo`
- on Android and iOS using `UNVERSIONED ncl` in `Expo Go`

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: apps/native-component-list/package.json [hunk 1]
```diff
@@ -170,7 +170,7 @@
     "expo-modules-tests-core": "^0.7.0",
     "url": "^0.11.0",
     "uuid": "^8.3.0",
-    "victory-native": "^30.4.0"
+    "victory-native": "^36.3.1"
```
Comment (tsapeta) [inline apps/native-component-list/package.json:173]: Is it necessary?

**Review by tsapeta — APPROVED** [review]
Some CI jobs are failing, but looks like they all are unrelated

File: apps/native-component-list/package.json (hunk 1 above)
Comment (bbarthec) [reply apps/native-component-list/package.json:173]: Previous version was written using deprecated React lifecycle methods and thus a warning was shown every time I've navigated to the screen consisting of the `victory-native`'s components.

**Comment by SimenB** [comment]
Hello! Would it be possible to upgrade to 12.4.1 (or later) for SDK 46? It includes https://github.com/react-native-svg/react-native-svg/pull/1806 which fixes https://github.com/react-native-svg/react-native-svg/issues/1755 causing issues with `@types/react@18`

**Pull request merged**